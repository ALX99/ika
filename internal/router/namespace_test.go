package router

import (
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

func Test_mergePaths(t *testing.T) {
	tests := []struct {
		name string // description of this test case
		// Named input parameters for target function.
		nsPath string
		path   string
	}{
		{
			name:   "root path join",
			nsPath: "/",
			path:   "/",
		},
		{
			name:   "normal path join",
			nsPath: "/test",
			path:   "/path",
		},
		{
			name:   "common base path",
			nsPath: "/test",
			path:   "/test/hi",
		},
		{
			name:   "with variables",
			nsPath: "/test/{something}",
			path:   "/test/hi/ok",
		},
		{
			name:   "with methods",
			nsPath: "POST /",
			path:   "/",
		},
		{
			name:   "with host",
			nsPath: "example.com:99/",
			path:   "/",
		},
		{
			name:   "with host and method",
			nsPath: "example.com/",
			path:   "GET /",
		},
		{
			name:   "with method and host",
			nsPath: "GET /example",
			path:   "example.com/",
		},
		{
			name:   "with mismatch methods",
			nsPath: "GET /example",
			path:   "POST /",
		},
		{
			name:   "with mismatch hosts",
			nsPath: "GET example.com/example",
			path:   "example1.com/",
		},
		{
			name:   "with matching hosts",
			nsPath: "GET example.com/example",
			path:   "example.com/",
		},
		{
			name:   "multiple slashes",
			nsPath: "GET //",
			path:   "//",
		},
		{
			name:   "tabbed path",
			nsPath: "TRACE /",
			path:   "TRACE\t/",
		},
		{
			name:   "spaces and tabs before path",
			nsPath: "GET /",
			path:   " \t /",
		},
		{
			name:   "path more specific",
			nsPath: "/",
			path:   "GET example.com/hi",
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got, err := mergePaths(tt.nsPath, tt.path)
			snaps.MatchSnapshot(t, got, err)
		})
	}
}
