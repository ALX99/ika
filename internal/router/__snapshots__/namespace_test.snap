
[Test_mergePaths/test_normal_path_join - 1]
/test/path
---

[Test_mergePaths/normal_path_join - 1]
/test/path
---

[Test_mergePaths/common_base_path - 1]
/test/test/hi
---

[Test_mergePaths/with_variables - 1]
/test/{something}/test/hi/ok
---

[Test_mergePaths/with_methods - 1]
/
---

[Test_mergePaths/root_path_join - 1]
/
---

[Test_mergePaths/with_host - 1]
example.com:99/
---

[Test_mergePaths/with_host_and_method - 1]
GET example.com/
---

[Test_mergePaths/with_method_and_host - 1]
/example/
---

[Test_mergePaths/with_mismatch_methods - 1]

---

[Test_mergePaths/with_mismatch_hosts - 1]

---

[Test_mergePaths/with_matching_hosts - 1]
example.com/example/
---

[Test_mergePaths/multiple_slashes - 1]
//
---
